x-pack/plugins/alerting/public/pages/maintenance_windows/components/create_maintenance_windows_form.tsx

```diff
@@ -62,317 +69,421 @@ const useDefaultTimezone = () => {
+    if (!Array.isArray(ruleTypes) || !Array.isArray(categoryIds) || !mounted) {
+      return [];
+    }
+
+    const featureIdsSet = new Set<ValidFeatureId>();
+
+    ruleTypes.forEach((ruleType) => {
+      if (categoryIds.includes(ruleType.category)) {
+        featureIdsSet.add(ruleType.producer as ValidFeatureId);
       }
-      if (!mounted) {
+    });
+
+    return [...featureIdsSet];
+  }, [ruleTypes, categoryIds, mounted]);
+
+  const onCategoryIdsChange = useCallback(
+    (ids: string[]) => {
+      if (!categoryIds) {
         return;
       }
-      if (hasSetInitialCategories.current) {
-        return;
+      setFieldValue('categoryIds', ids);
+    },
+    [categoryIds, setFieldValue]
+  );
+
+  const onScopeQueryToggle = useCallback(
+    (isEnabled: boolean) => {
+      if (isEnabled) {
+        setFieldValue('categoryIds', [categoryIds?.sort()[0] || availableCategories.sort()[0]]);
       }
-      if (Array.isArray(categoryIds)) {
-        return;
+      setIsScopedQueryEnabled(isEnabled);
+    },
+    [categoryIds, availableCategories, setFieldValue]
+  );
+
+  const onQueryChange = useCallback(
+    (newQuery: string) => {
+      if (isQueryInvalid) {
+        setIsQueryInvalid(false);
       }
-      setFieldValue('categoryIds', [
-        DEFAULT_APP_CATEGORIES.observability.id,
-        DEFAULT_APP_CATEGORIES.security.id,
-        DEFAULT_APP_CATEGORIES.management.id,
-      ]);
-      hasSetInitialCategories.current = true;
-      // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [isEditMode, categoryIds, mounted]);
-
-    return (
-      <Form form={form}>
-        <EuiFlexGroup direction="column" responsive={false}>
-          <EuiFlexItem>
-            <UseField
-              path="title"
-              componentProps={{
-                'data-test-subj': 'title-field',
-                euiFieldProps: {
-                  autoFocus: true,
-                },
-              }}
-            />
-          </EuiFlexItem>
-          <EuiSpacer size="xs" />
-          <EuiFlexItem>
-            <EuiText size="s">
-              <h4>{i18n.CREATE_FORM_TIMEFRAME_TITLE}</h4>
-              <p>
-                <EuiTextColor color="subdued">
-                  {i18n.CREATE_FORM_TIMEFRAME_DESCRIPTION}
-                </EuiTextColor>
-              </p>
-            </EuiText>
-          </EuiFlexItem>
-          <EuiFlexItem>
-            <EuiFlexGroup alignItems="flexEnd" responsive={false}>
-              <EuiFlexItem grow={3}>
-                <UseMultiFields
-                  fields={{
-                    startDate: {
-                      path: 'startDate',
-                      config: {
-                        label: i18n.CREATE_FORM_SCHEDULE,
-                        defaultValue: defaultStartDateValue,
-                        validations: [],
-                      },
+      setQuery(newQuery);
+    },
+    [isQueryInvalid]
+  );
+
+  const modal = useMemo(() => {
+    let m;
```

Comment:
I thought that jsx, we will not like to render an undefined value but it will be ok with `null`.

Reply:
react is too smart

Reply:
maybe it just got smarter!